test/model.populate.test.js

```diff
@@ -1946,165 +1578,64 @@ describe('model: populate:', function() {
 
     const alice = new User({ _id: 2359, name: 'Alice' });
 
-    alice.save(function(err) {
-      assert.ifError(err);
-
-      const note = new Note({ author: 2359, body: 'Buy Milk' });
-      note.save(function(err) {
-        assert.ifError(err);
-
-        Note.findById(note.id).populate('author').exec(function(err, note) {
-          assert.ifError(err);
-          assert.equal(note.body, 'Buy Milk');
-          assert.ok(note.author);
-          assert.equal(note.author.name, 'Alice');
-          done();
-        });
-      });
-    });
-  });
-
-  it('required works on ref fields (gh-577)', function(done) {
```

Comment:
You're right, not sure why this one was removed. Likely because we assumed it was redundant and hard to refactor to async/await. But turns out it isn't redundant.